src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/IConfigurationSection.cs

```diff
@@ -21,6 +21,6 @@ public interface IConfigurationSection : IConfiguration
         /// <summary>
         /// Gets or sets the section value.
         /// </summary>
-        string Value { get; set; }
+        string? Value { get; set; }
```

Comment:
if you decide to keep path as nullable arg in other places should this also accept nullable path?

Reply:
I don't think so. `Path` will never be set to `null`. All sections have a path.

Also all usages don't expect null in `Path`:
- https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationRootExtensions.cs#L28
- https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationExtensions.cs#L57
- https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationExtensions.cs#L64

Regarding leaving `path` parameters us null. I'm not objecting removing `?`. My only concern is that it will be harder to call those methods by requiring additional null-check on their side.